File: frontend/src/types/rewards.ts
/** Reward types — mirrors backend schemas/rewards.py. */

export interface Badge {
  code: string;
  title: string;
  description: string | null;
  earned: boolean;
  earned_at: string | null;
}

export interface RewardEvent {
  id: number;
  event_type: string;
  points: number;
  description: string | null;
  related_entity_type: string | null;
  related_entity_id: number | null;
  created_at: string;
}

export interface RewardsSummary {
  user_id: number;
  total_points: number;
  current_streak: number;
  longest_streak: number;
  badge_count: number;
  badges: Badge[];
  recent_events: RewardEvent[];
}

File: backend/app/schemas/rewards.py
"""Reward request/response schemas (spec 14 rewards section)."""

from datetime import datetime

from pydantic import BaseModel


class BadgeOut(BaseModel):
    code: str
    title: str
    description: str | None
    earned: bool
    earned_at: datetime | None


class RewardEventOut(BaseModel):
    id: int
    event_type: str
    points: int
    description: str | None
    related_entity_type: str | None
    related_entity_id: int | None
    created_at: datetime


class RewardsSummaryOut(BaseModel):
    user_id: int
    total_points: int
    current_streak: int
    longest_streak: int
    badge_count: int
    badges: list[BadgeOut]  # earned, most recent first
    recent_events: list[RewardEventOut]
